unskript-ctl/unskript_ctl_factory.py

```diff
@@ -258,4 +261,53 @@ def get_global(self):
         return self._get('global')
     
     def get_checks_params(self):
-        return self._get('checks', 'arguments')
\ No newline at end of file
+        return self._get('checks', 'arguments')
+    
+
+
+# This class is a simple schema validation. we define a mandatory fields
+# that need to be present in the Yaml file. If there is a new field that needs
+# to be added, ensure, it is added to the mandatory_fields and expected_order
+# too. 
+class SchemaValidator(UnskriptFactory):
+    def __init__(self, **kwargs):
+        self.logger.debug("Initializing SchemaValidator class")
+
+        # Defined as an ordered list with type information
+        self.expected_structure = [
+            ('version', str),
```

Comment:
This is not sufficient, it only handles top level keys, how about nested keys? You need to define classes for each top level as well. 

Having this expected_structure doesnt look right to me, why not define a class and convert the json to an object of that class? I think unctl does something similar.